resources/prosody-plugins/token/util.lib.lua

```diff
@@ -1,11 +1,108 @@
 -- Token authentication
 -- Copyright (C) 2015 Atlassian
 
+local basexx = require "basexx";
+local have_async, async = pcall(require, "util.async");
+local hex = require "util.hex";
+local http = require "net.http";
+local json = require "cjson";
+local path = require "util.paths";
+local sha256 = require "util.hashes".sha256;
+local timer = require "util.timer";
+
 local jwt = require "luajwtjitsi";
 
 local _M = {};
 
-local function _verify_token(token, appId, appSecret, disableRoomNameConstraints)
+local appId = module:get_option_string("app_id");
+local appSecret = module:get_option_string("app_secret");
+local asapKeyServer = module:get_option_string("asap_key_server");
+local allowEmptyToken = module:get_option_boolean("allow_empty_token");
+local disableRoomNameConstraints = module:get_option_boolean("disable_room_name_constraints");
+
+-- TODO: Figure out a less arbitrary default cache size.
+local cacheSize = module:get_option_number("jwt_pubkey_cache_size", 128);
+local cache = require"util.cache".new(cacheSize);
+
+local function _check_config()
+    if allowEmptyToken == true then
+        module:log("warn", "WARNING - empty tokens allowed");
+    end
+
+    if appId == nil then
+        module:log("error", "'app_id' must not be empty");
+        return true;
+    end
+
+    if appSecret == nil and asapKeyServer == nil then
+        module:log("error", "'app_secret' or 'asap_key_server' must be specified");
+        return true;
+    end
+
+    if asapKeyServer and not have_async then
+        module:log("error", "requires a version of Prosody with util.async");
+        return true;
+    end
+
+    return false;
+end
+
+local http_timeout = 30;
```

Comment:
I know it was like that before, but could you please move "http_timeout" and "http_headers" vars declarations to the top, next to the others ?